backend/src/services/integrationService.ts

```diff
@@ -133,27 +135,88 @@ export default class IntegrationService {
     return IntegrationRepository.findAllByPlatform(platform, this.options)
   }
 
+  static isCodePlatform(value: string): value is CodePlatform {
+    return [
+      PlatformType.GITHUB,
+      PlatformType.GITHUB_NANGO,
+      PlatformType.GITLAB,
+      PlatformType.GIT,
+      PlatformType.GERRIT,
+    ].includes(value as PlatformType)
+  }
+
   async create(data, transaction?: any, options?: IRepositoryOptions) {
     try {
       const record = await IntegrationRepository.create(data, {
         ...(options || this.options),
         transaction,
       })
 
+      const collectionService = new CollectionService(this.options)
+
+      const [insightsProject] = await collectionService.findInsightsProjectsBySegmentId(
+        record.segmentId,
+      )
+
+      if (insightsProject) {
+        this.updateInsightsProject(insightsProject.id, record.segmentId)
+      }
+
       return record
     } catch (error) {
       SequelizeRepository.handleUniqueFieldError(error, this.options.language, 'integration')
       throw error
     }
   }
 
+  private async updateInsightsProject(insightsProjectId: string, segmentId: string) {
+    const collectionService = new CollectionService(this.options)
+
+    const data: Partial<ICreateInsightsProject> = {}
+    const { platforms, widgets } = await collectionService.findSegmentsWidgetsById(segmentId)
+    data.widgets = widgets
+
+    if (platforms.some(IntegrationService.isCodePlatform)) {
+      const repositories = await collectionService.findRepositoriesForSegment(segmentId)
+      data.repositories = [
+        ...new Set(Object.values(repositories).flatMap((entries) => entries.map((e) => e.url))),
+      ]
+    }
+
+    if (platforms.includes(PlatformType.GITHUB)) {
+      const githubInsights = await collectionService.findGithubInsightsForSegment(segmentId)
+      if (githubInsights) {
+        data.description = githubInsights.description
+        data.github = githubInsights.github
+        data.logoUrl = githubInsights.logoUrl
+        data.name = githubInsights.name
+        data.twitter = githubInsights.twitter
+        data.website = githubInsights.website
+      }
+    }
+
+    this.options.log.info(`Insight Project updated: ${insightsProjectId}`)
+
+    await collectionService.updateInsightsProject(insightsProjectId, data)
+  }
+
   async update(id, data, transaction?: any, options?: IRepositoryOptions) {
     try {
       const record = await IntegrationRepository.update(id, data, {
         ...(options || this.options),
         transaction,
       })
 
+      const collectionService = new CollectionService(this.options)
+
+      const [insightsProject] = await collectionService.findInsightsProjectsBySegmentId(
+        record.segmentId,
+      )
+
+      if (insightsProject) {
+        this.updateInsightsProject(insightsProject.id, record.segmentId)
```

Comment:
If it's an "update integrations" we probably should have some sort of flag passed to "updateInsightsProject" from preventing to run `findGithubInsightsForSegment` every time we update an integration. We should only call this method if the github integration was created/updated.